# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

**Lane:** Advanced A2 (Growth / Recovery / Momentum Prediction)

**Why:** Standard classification models act retroactively—flagging a page only after traffic is already lost. I want to build a time-aware predictive model (using forward-window validation) that identifies "at-risk" pages before the decline happens, shifting the team from reactive to proactive.

## 2. The question: decision, action, cost of a wrong call

- **Decision:** Which currently stable or healthy pages are at the highest statistical risk of declining over the next 30 days?
- **Action:** The content team will proactively refresh the content, update metadata, or build internal links to protect the page's rankings.
- **Cost of a wrong call:** A *false positive* means wasting 2-5 hours of a writer's time refreshing a page that didn't actually need it. A *false negative* means losing organic traffic (and potentially revenue) that could have been saved with a simple, timely update.

## 3. Quick look at the data (2-3 real numbers)

To justify a forward-looking predictive model, we need to show that content decay is systemic and fast-acting. I will examine the starter dataset to find out what percentage of our content inventory is currently decaying, and what the median age of a decaying piece of content is.

In [1]:
import pandas as pd

df = pd.read_csv("../../data/raw/content_refresh_anonymized.csv")

total_pages = len(df)
declining_pages = len(df[df["trend_direction"] == "down"])
pct_declining = (declining_pages / total_pages) * 100
median_age_declining = df[df["trend_direction"] == "down"]["content_age_days"].median()

print(f"Total Pages: {total_pages:,}")
print(f"Pages Currently Declining: {declining_pages:,} ({pct_declining:.1f}%)")
print(f"Median Age of Declining Pages: {median_age_declining} days (approx. {median_age_declining/30:.1f} months)")

Total Pages: 30,000
Pages Currently Declining: 16,262 (54.2%)
Median Age of Declining Pages: 216.0 days (approx. 7.2 months)


**Observation:** Over 54% of the pages in the starter slice are already declining, meaning decay is a massive systemic issue. Furthermore, the median age of a declining page is only ~216 days (7 months). Because decay happens so quickly, relying on retroactive flags (waiting until the traffic is already gone) is too slow. We need a 30-day forward-looking predictive window to intercept these pages before the 7-month mark.

## 4. Careful words: what I can and can't claim

- **What I can claim:** I can build a model that predicts the statistical probability of a future traffic drop based on historical observable signals. I can also claim this model is validated honestly using a strict time-series split to prevent data leakage.
- **What I can't claim:** I cannot claim I have reverse-engineered Google's algorithm. Furthermore, because this is an observational risk-scoring model and not a causal experiment, I cannot guarantee with 100% certainty that refreshing the page *will* prevent the decline.

## Self-check

Before you submit, confirm each line honestly:

- [x] Every section above is filled — markdown thinking AND the code that backs it
- [x] The notebook runs top to bottom with no errors (Runtime → Run all)
- [x] No client names, URLs, or private queries anywhere
- [x] My claims use careful words: observed, measured, directional, decision-support
- [x] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.